src/factor_graph.cc

```diff
@@ -43,38 +43,49 @@ std::ostream &operator<<(std::ostream &stream,
   return stream;
 }
 
-template <class T>
-T *fast_alloc(size_t num) {
-  void *raw_memory = operator new[](num * sizeof(T));
-  return static_cast<T *>(raw_memory);
-}
-
 FactorGraph::FactorGraph(const FactorGraphDescriptor &capacity)
     : capacity(capacity),
       size(),
       // fast alloc: 0 sec for a 270M-factor graph
-      weights(fast_alloc<Weight>(capacity.num_weights)),
-      factors(fast_alloc<Factor>(capacity.num_factors)),
-      vifs(fast_alloc<FactorToVariable>(capacity.num_edges)),
-      variables(fast_alloc<Variable>(capacity.num_variables)),
-      factor_index(fast_alloc<size_t>(capacity.num_edges)),
-      values(fast_alloc<VariableToFactor>(capacity.num_values))
+      weights(fast_alloc_no_init<Weight>(capacity.num_weights)),
+      factors(fast_alloc_no_init<Factor>(capacity.num_factors)),
+      vifs(fast_alloc_no_init<FactorToVariable>(capacity.num_edges)),
+      variables(fast_alloc_no_init<Variable>(capacity.num_variables)),
+      factor_index(fast_alloc_no_init<size_t>(capacity.num_edges)),
+      values(fast_alloc_no_init<VariableToFactor>(capacity.num_values))
 // slow alloc: 55 sec for a 270M-factor graph
 // weights(new Weight[capacity.num_weights]),
 // factors(new Factor[capacity.num_factors]),
 // vifs(new FactorToVariable[capacity.num_edges]),
 // variables(new Variable[capacity.num_variables]),
 // factor_index(new size_t[capacity.num_edges]),
 // values(new VariableToFactor[capacity.num_values])
-{}
+{
+  for (size_t i = 0; i < capacity.num_variables; ++i) {
+    // make sure the pointers are null after the fast mem allocation
+    // NOTE: if we introduce more pointers in the data structures, add here.
+    // see also ~FactorGraph()
+    variables[i].domain_map.release();
+    variables[i].adjacent_factors.release();
+  }
+}
```

Comment:
I'm curious how much time this adds to the 0 secs?  Would be nice to leave a comment line if you had measured already. :)